esx_service/vmdk_ops.py

```diff
@@ -1711,6 +1712,101 @@ def wait_for_tasks(si, tasks):
 
 #------------------------
 
+def start_vm_changelistener():
+    """
+    Listen to power state changes of VMs running on current host
+    """
+    si = get_si()
+    pc = si.content.propertyCollector
+    create_vm_powerstate_filter(pc, si.content.rootFolder)
```

Comment:
Done. Capturing the return value of this function. But I think it would be right to start the main service in spite of any error while starting the VM listener.